### PR TITLE
Refactor Autosuggest

### PR DESCRIPTION
**Pull request by goremikins**
## Description
Apply minor refactor to remove arrow functions and use references.

Fixes #
ELF-1494
ELF-1778

## Type of change
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Refactor Code

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/elements/src/autosuggest/index.ts [hunk 1]
```diff
@@ -1404,7 +1378,63 @@ export class Autosuggest extends Overlay {
...
+   * @returns template of loader if currently query loading
+   */
+  protected get loaderTemplate (): TemplateResult | null {
+    if (!this.loading) {
+      return null;
+    }
+    return html`
+      <div part="loader">
+        <div part="backdrop"></div>
+        <ef-loade></ef-loader>
```
Comment (wsuwt) [inline packages/elements/src/autosuggest/index.ts]:
```suggestion
        <ef-loader aria-label="Loading" aria-live="assertive"></ef-loader>
```

Comment (wsuwt) [reply packages/elements/src/autosuggest/index.ts]: Is there anything we need to do to make this loader accessible ?

Comment (wsuwt) [reply packages/elements/src/autosuggest/index.ts:1393]: Maybe something like [this](https://v6.refinitiv-ui.pages.dev/elements/loader#accessibility) would do.

Comment (goremikins) [reply packages/elements/src/autosuggest/index.ts]: OK, I will do it as part of Accessibility PR

**Review by Sakchai-Refinitiv — COMMENTED** [review]
This change looks good, waiting for https://github.com/Refinitiv/refinitiv-ui/pull/338 merged to this PR.

**Review by Sakchai-Refinitiv — DISMISSED** [review]

**Review by wsuwt — APPROVED** [review]

**Pull request merged**